src/EStimLibrary/Core/ResourceManager.cs

```diff
@@ -63,7 +65,7 @@ public bool TryAddResource(int globalId, ResourceType resource)
         this.Resources.Add(globalId, resource);
         // small TODO: the Use method calls IsValidId again --> any way to
         // refactor to make more efficient?
-        this.IdPool.UseId(globalId);
+        this.IdPool.UsedIds.Add(globalId);
```

Comment:
Was UseId() not adding it correctly? Similar thing: if that wasn't working, should fix that rather than work around it here. Or was there a different motivation for the change here?

Reply:
I have reverted the changes and it passes the test cases